src/lib/common/sol-platform-linux-common.c

```diff
@@ -776,3 +798,99 @@ sol_platform_linux_uevent_unsubscribe(const char *action, const char *subsystem,
 
     return 0;
 }
+
+
+int64_t
+sol_platform_impl_get_system_clock(void)
+{
+    return (int64_t)time(NULL);
+}
+
+static bool
+system_clock_changed(void *data, int fd, uint32_t active_flags)
+{
+    char buf[4096];
+
+    if (read(timer_ctx.fd, buf, sizeof(buf)) >= 0)
```

Comment:
This is by design. If the execution reach this code it means that the system clock changed and I need to register a new one.